src/analysis/processing/qgsalgorithmrasterize.cpp

```diff
@@ -150,6 +151,72 @@ QVariantMap QgsRasterizeAlgorithm::processAlgorithm( const QVariantMap &paramete
   int height { yTileCount * tileSize };
   int nBands { transparent ? 4 : 3 };
 
+  int64_t totalTiles = 0;
+  for ( auto &layer : std::as_const( mMapLayers ) )
+  {
+    if ( QgsRasterLayer *rasterLayer = qobject_cast<QgsRasterLayer *>( ( layer.get() ) ) )
+    {
+      if ( const QgsProviderMetadata *metadata = rasterLayer->providerMetadata() )
+      {
+        QVariantMap details = metadata->decodeUri( rasterLayer->source() );
+        if ( details.contains( QStringLiteral( "url" ) ) && QUrl( details[QStringLiteral( "url" )].toString() ).host() == QStringLiteral( "tile.openstreetmap.org" ) )
+        {
+          const QList< double > resolutions = rasterLayer->dataProvider()->nativeResolutions();
+          if ( resolutions.isEmpty() )
+          {
+            continue;
+          }
+
+          if ( totalTiles == 0 )
+          {
+            const QgsCoordinateTransform ct( context.project()->crs(), rasterLayer->crs(), context.transformContext() );
+            QgsRectangle extentLayer;
+            try
+            {
+              extentLayer = ct.transform( extent );
+            }
+            catch ( QgsCsException & )
+            {
+              totalTiles = -1;
+              continue;
+            }
+
+            const double mapUnitsPerPixelLayer = extentLayer.width() / width;
+            int i;
+            for ( i = 0; i < resolutions.size() && resolutions.at( i ) < mapUnitsPerPixelLayer; i++ )
+            {
+            }
+
+            if ( i == resolutions.size() ||
+                 ( i > 0 && resolutions.at( i ) - mapUnitsPerPixelLayer > mapUnitsPerPixelLayer - resolutions.at( i - 1 ) ) )
+            {
+              i--;
+            }
+
+            const int nbTilesWidth = std::ceil( extentLayer.width() / resolutions.at( i ) / 256 );
+            const int nbTilesHeight = std::ceil( extentLayer.height() / resolutions.at( i ) / 256 );
+            totalTiles = nbTilesWidth * nbTilesHeight;
```

Comment:
I had anticipated the clang_tidy warning ;-)

```suggestion
            totalTiles = static_cast< qint64_t >(nbTilesWidth) * nbTilesHeight;
```